plugins/needs_restarting.py

```diff
@@ -303,19 +305,32 @@ def run(self):
                 return None
 
         stale_pids = set()
+        stale_service_names = set()
         uid = os.geteuid() if self.opts.useronly else None
         for ofile in list_opened_files(uid):
             pkg = owning_pkg_fn(ofile.presumed_name)
+            pid = ofile.pid
             if pkg is None:
                 continue
-            if pkg.installtime > process_start(ofile.pid):
-                stale_pids.add(ofile.pid)
+            if pkg.installtime <= process_start(pid):
+                continue
+            if self.opts.services or self.opts.exclude_services:
+                service_name = get_service_dbus(pid)
+                if service_name is None:
+                    stale_pids.add(pid)
+                else:
+                    stale_service_names.add(service_name)
+                    if not self.opts.exclude_services:
+                        stale_pids.add(pid)
```

Comment:
I didn't realize this before but now that the `--services` option prints only `stale_service_names` there is no point in adding to the `stale_pids` when `--services` is set right?

Reply:
I suppose, but I think it's fine to leave it.